In [22]:
# # Auxiliares ao desenvolvimento, remover uma vez que o código está funcional
%load_ext autoreload
%autoreload 2
%reload_ext autoreload

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [60]:
import os
from maquinas.dados import Dados

dados = Dados.ler_dados(nome_arquivo='../dados/dados.txt', separador=",")


dados

{'vn_a': '240',
 'in_a': '10',
 'ra': '2',
 'rf': '120',
 'reos': '0-100',
 'ke': '0.8',
 'kt': '0.8',
 'iner': '0.02',
 'wn': '1800',
 'pn': '2.5',
 'n': '85',
 'torq': '8-15'}

In [53]:
for d in dados.items():
    print (d)

('Tensão de armadura nominal', ' 240 V')
('Corrente nominal de armadura', ' 10 A')
('Resistência de armadura (Ra)', ' 2 Ω')
('Resistência de campo (Rf )', ' 120 Ω')
('Reostato de campo', ' 0 a 100 Ω')
('Constantes eletromecânicas (Ke, Kt)', ' 0,8')
('Momento de inércia total', ' 0,02 kg·m2')
('Velocidade nominal', ' 1800 rpm')
('Potência nominal', ' 2,5 kW')
('Rendimento', ' 85%')
('Torque de carga', ' entre 8 e 15 N·m')


In [ ]:
from maquinas.motor_cc_shunt import MotorCCShunt
from maquinas.dados import Dados


motor = MotorCCShunt(dados={})


Objeto MotorCCShunt criado...
Dados:
 	 {}


(10, 40)

 $ entre 8 e 15 N\cdot m $ 

# Fluxo de trabalho geral

# Fluxo de trabalho detalhado

1. Criar estruturas para leitura e escrita de arquivos
   1. Essas estruturas devem ler os dados de forma padronizada e retornar um dicionário para posterior uso
      1. O dicionário pode conter campos para comentário
2. Criar uma classe que consuma os dados lidos dos arquivos
3. 

# Condições de operação

## Condição 1: Operação nominal (240 V, sem reostato, carga nominal);
## Condição 2: Subtensão (200 V, carga nominal);
## Condição 3: Enfraquecimento de campo (reostato = 50 Ω);
## Condição 4: Baixa carga (torque reduzido);
## Condição 5: Condição crítica (baixa tensão e alto torque).

In [25]:
#